### PR TITLE
Lint code

### PR DESCRIPTION
**Pull request by kelonye**
Ref Teamweek/teamweek#2510

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ],
+  "env": {
+    "browser": true,
+    "commonjs": true,
+    "es6": true,
+    "node": true
+  },
+  "globals": {
+    "chrome": true
+}
```
Comment (herkyl) [inline .eslintrc.json]: JSON formatting issue

File: src/background/injector.js [hunk 2]
```diff
@@ -1,48 +1,47 @@
...
+  initialize() {
     this.listeners = {};
 
     this.collection = new CustomDomainCollection();
 
     this.collection.on('add', (model) => this.addNavigationListener(model));
-    this.collection.on('remove', (model) => this.removeNavigationListener(model));
-    
+    this
+      .collection.on('remove', (model) => this.removeNavigationListener(model));
```
Comment (herkyl) [inline src/background/injector.js]: Remove the brackets to fit to a single line
```js
    this.collection.on('remove', model => this.removeNavigationListener(model));
```

File: src/models/tokens_model.js [hunk 3]
```diff
@@ -17,8 +16,8 @@ var TokensModel = Model.extend({
     /** Whether the access and refresh tokens are set */
     has_auth_tokens: {
       deps: ['access_token', 'refresh_token'],
-      fn: function() {
-        return this.access_token != null && this.refresh_token != null;
+      fn() {
+        return !!this.access_token && !!this.refresh_token;
```
Comment (herkyl) [inline src/models/tokens_model.js]: `return this.access_token && this.refresh_token;` will suffice

File: src/popup/fields/estimate_field.js [hunk 4]
```diff
@@ -12,19 +11,19 @@ var EstimateField = View.extend({
   derived: {
     isFilled: {
       deps: ['raw'],
-      fn: function() {
-        return this.raw != null && this.raw.length > 0;
+      fn() {
+        return !!this.raw && this.raw.length > 0;
```
Comment (herkyl) [inline src/popup/fields/estimate_field.js]: `this.raw && this.raw.length > 0;` will cast it to bool already

File: src/popup/fields/text_field.js [hunk 5]
```diff
@@ -1,16 +1,15 @@
...
     value: 'string'
   },
 
   derived: {
     isFilled: {
       deps: ['value'],
-      fn: function() {
-        return this.value != null && this.value.length > 0;
+      fn() {
+        return !!this.value && this.value.length > 0;
```
Comment (herkyl) [inline src/popup/fields/text_field.js]: `return this.value && this.value.length > 0;` will suffice

File: Gulpfile.js [hunk 6]
```diff
@@ -1,7 +1,6 @@
-var path = require('path');
-var gulp = require('gulp');
-var plugins = require('gulp-load-plugins')();
-var merge = require('merge-stream');
+const gulp = require('gulp');
+const plugins = require('gulp-load-plugins')();
+const merge = require('merge-stream');
```
Comment (dhruvparmar372) [inline Gulpfile.js]: introducing ES6 keywords means project probably won't run on `node < 6`. 
let's add that information in `package.json` under `engines` property.

Comment (kelonye) [reply Gulpfile.js]: good idea 👍🏽

**Pull request merged**